corehq/blobs/s3db.py

```diff
@@ -41,73 +42,122 @@ def __init__(self, config):
+                self.metadb.put(meta)
+            else:
+                # legacy: can be removed with old API
+                timeout = blob_meta_args.get("timeout")
+                content_md5 = get_content_md5(content)
+                content_length = get_file_size(content)
+                if timeout is not None:
+                    set_blob_expire_object(bucket, identifier, content_length, timeout)
+                datadog_counter('commcare.blobs.added.count')
+                datadog_counter('commcare.blobs.added.bytes', value=content_length)
+                meta = BlobInfo(identifier, content_length, "md5-" + content_md5)
+            with self.report_timing('put', key):
+                s3_bucket.upload_fileobj(content, key)
+        return meta
+
+    def get(self, identifier=None, bucket=DEFAULT_BUCKET, key=None):
+        if not (identifier is None and bucket == DEFAULT_BUCKET):
+            # legacy: can be removed with old API
+            assert key is None, key
+            key = self.get_path(identifier, bucket)
+        check_safe_key(key)
+        with maybe_not_found(throw=NotFound(key)), self.report_timing('get', key):
+            resp = self._s3_bucket().Object(key).get()
+        return BlobStream(resp["Body"], self, key)
+
+    def size(self, identifier=None, bucket=DEFAULT_BUCKET, key=None):
+        if not (identifier is None and bucket == DEFAULT_BUCKET):
+            # legacy: can be removed with old API
+            assert key is None, key
+            key = self.get_path(identifier, bucket)
+        check_safe_key(key)
+        with maybe_not_found(throw=NotFound(key)), self.report_timing('size', key):
+            return self._s3_bucket().Object(key).content_length
+
+    def exists(self, identifier=None, bucket=DEFAULT_BUCKET, key=None):
+        if not (identifier is None and bucket == DEFAULT_BUCKET):
+            # legacy: can be removed with old API
+            assert key is None, key
+            key = self.get_path(identifier, bucket)
+        check_safe_key(key)
         try:
-            with maybe_not_found(throw=NotFound(identifier, bucket)), \
-                    self.report_timing('exists', identifier, bucket):
-                self._s3_bucket().Object(path).load()
+            with maybe_not_found(throw=NotFound(key)), self.report_timing('exists', key):
+                self._s3_bucket().Object(key).load()
             return True
         except NotFound:
             return False
 
     def delete(self, *args, **kw):
-        identifier, bucket = self.get_args_for_delete(*args, **kw)
-        path = self.get_path(identifier, bucket)
+        if "key" in kw:
+            assert set(kw) == {"key"} and not args, (args, kw)
+            key = kw["key"]
+        else:
+            # legacy: can be removed with old API
+            identifier, bucket = self.get_args_for_delete(*args, **kw)
+            key = self.get_path(identifier, bucket)
```

Comment:
this `if / else` doesn't seem necessary since you have it duplicated below

Reply:
https://github.com/dimagi/commcare-hq/pull/21569/commits/00a3ec107a9b963878f8cf5be5e8e176fc6b522d